src/Domain/OrientationMap.hpp

```diff
@@ -140,6 +140,18 @@ std::array<tt::remove_cvref_wrap_t<T>, VolumeDim> discrete_rotation(
     std::array<T, VolumeDim> source_coords) noexcept {
   using ReturnType = tt::remove_cvref_wrap_t<T>;
   std::array<ReturnType, VolumeDim> new_coords{};
+  ASSERT(
+      [&rotation]() {
+        std::set<size_t> set_of_dims;
+        for (size_t j = 0; j < VolumeDim; j++) {
+          set_of_dims.insert(
+              rotation(gsl::at(Direction<VolumeDim>::all_directions(), j))
+                  .dimension());
+        }
+        return set_of_dims.size() == VolumeDim;
```

Comment:
There are now three (near) copies of this block throughout the code.  Put it in a method or something.

Reply:
called something like `is_valid_rotation` or `is_valid_orientation_map`?